pkg/cmd/pulumi/state/state_reprotect.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2016-2025, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package state
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/pulumi/pulumi/pkg/v3/backend/display"
+	"github.com/pulumi/pulumi/pkg/v3/cmd/pulumi/backend"
+	"github.com/pulumi/pulumi/pkg/v3/resource/deploy"
+	"github.com/pulumi/pulumi/pkg/v3/resource/edit"
+	pkgWorkspace "github.com/pulumi/pulumi/pkg/v3/workspace"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/env"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
+
+	"github.com/spf13/cobra"
+)
+
+func newStateReprotectCommand() *cobra.Command {
+	var protectAll bool
+	var stack string
+	var yes bool
+
+	cmd := &cobra.Command{
+		Use:   "reprotect [resource URN]",
+		Short: "Re-protect (undo a `pulumi unprotect`) resources in a stack's state",
+		Long: `Re-Protect resource in a stack's state
```

Comment:
As discussed, protect, not reprotect, and we should amend the docs etc. accordingly.

Reply:
Did we discuss and reject the `pulumi state unprotect --undo` proposal? Should we discuss in ideation?